packages/@aws-cdk/aws-redshift/README.md

```diff
@@ -386,3 +386,46 @@ new Cluster(stack, 'Redshift', {
 ```
 
 If enhanced VPC routing is not enabled, Amazon Redshift routes traffic through the internet, including traffic to other services within the AWS network.
+
+## Default IAM role
+
+Some Amazon Redshift features require Amazon Redshift to access other AWS services on your behalf. For your Amazon Redshift clusters to act on your behalf, you supply security credentials to your clusters. The preferred method to supply security credentials is to specify an AWS Identity and Access Management (IAM) role.
+
+When you create an IAM role and set it as the default for the cluster using console, you don't have to provide the IAM role's Amazon Resource Name (ARN) to perform authentication and authorization.
+
+```ts
+declare const vpc: ec2.Vpc;
+
+const defaultRole = new iam.Role(this, 'DefaultRole', {
+  assumedBy: new iam.ServicePrincipal('redshift.amazonaws.com'),
+},
+);
+
+new Cluster(stack, 'Redshift', {
+    masterUser: {
+      masterUsername: 'admin',
+    },
+    vpc,
+    defaultRole,
+});
+```
+
+A default role can also be added to a cluster using the `addDefaultIamRole` method.
```

Comment:
Might be misunderstanding but can this be added to an already existing cluster? If so is there a possibility this default role could attempt to append an 11th role to an existing cluster and then go over 10?

Reply:
The way this is set up it could be added to an already existing cluster which would push the number of IAM roles to more than 10. It's interesting that this limit has been imposed as for most regions the [role limit for a cluster is 50.](https://docs.aws.amazon.com/redshift/latest/mgmt/amazon-redshift-limits.html#:~:text=Cluster%20IAM%20roles%20for%20Amazon%20Redshift%20to%20access%20other%20AWS%20services) Is there a way to make this limit region specific?

I'll try to find a way to check the total number of roles before adding the default role.

Reply:
It's also interesting that there doesn't seem to be a check for the IAM role list to make sure there's less than 10 - should this be added?

Reply:
I am thinking of changing the functionality so that the default role needs to be defined in the roles parameter on cluster creation, or already attached to the cluster if adding post-creation. This would remove the need to check for the total number of IAM roles and I think would reflect better practice.
What are your thoughts?